ppb_vector/vector2.py

```diff
@@ -114,4 +114,6 @@ def reflect(self, surface_normal: 'Vector2') -> 'Vector2':
         """
         if not (0.99999 < surface_normal.length < 1.00001):
             raise ValueError("Reflection requires a normalized vector.")
+        if self * surface_normal > 0:
+            self = -self
```

Comment:
This is also a thing I would rather we avoid: self should always refer to the class being operated on, reassigning to self is a smell.

Can we add a separate variable for this?